app/screens/home/recent_mentions/components/mention/mention.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {View, StyleSheet} from 'react-native';
+
+import Avatar from '@components/post_list/post/avatar';
+import Message from '@components/post_list/post/body/message';
+import Header from '@components/post_list/post/header';
+import SystemAvatar from '@components/system_avatar';
+import SystemHeader from '@components/system_header';
+import {Screens} from '@constants';
+import {useTheme} from '@context/theme';
+import {fromAutoResponder, isFromWebhook, isSystemMessage, isEdited as postEdited} from '@utils/post';
+
+import ChannelInfo from '../channel_info';
+
+import type PostModel from '@typings/database/models/servers/post';
+import type UserModel from '@typings/database/models/servers/user';
+
+type Props = {
+    currentUser: UserModel;
+    post: PostModel;
+}
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        paddingHorizontal: 20,
+    },
+    content: {
+        flexDirection: 'row',
+        paddingBottom: 8,
+    },
+    rightColumn: {
+        flex: 1,
+        flexDirection: 'column',
+        marginRight: 12,
+    },
+    message: {
+        flex: 1,
+    },
+});
+
+function Mention({post, currentUser}: Props) {
+    const theme = useTheme();
+
+    const isAutoResponder = fromAutoResponder(post);
+    const isSystemPost = isSystemMessage(post);
+    const isWebHook = isFromWebhook(post);
+    const isEdited = postEdited(post);
+
+    const postAvatar = isAutoResponder ? (
+        <SystemAvatar theme={theme}/>
+    ) : (
+        <Avatar
+            isAutoReponse={isAutoResponder}
+            isSystemPost={isSystemPost}
+            post={post}
+        />
+    );
+
+    const header = isSystemPost && !isAutoResponder ? (
+        <SystemHeader
+            createAt={post.createAt}
+            theme={theme}
+        />
+    ) : (
+        <Header
+            currentUser={currentUser}
+            isAutoResponse={isAutoResponder}
+            differentThreadSequence={true}
+            isEphemeral={false}
+            isPendingOrFailed={false}
+            isSystemPost={isSystemPost}
+            isWebHook={isWebHook}
+            location={Screens.MENTIONS}
+            post={post}
+            shouldRenderReplyButton={false}
+        />
+    );
+
+    return (
+        <View style={styles.container}>
+            <ChannelInfo post={post}/>
+            <View style={styles.content}>
+                {postAvatar}
+                <View style={styles.rightColumn}>
+                    {header}
+                    <View style={styles.message}>
+                        <Message
```

Comment:
it appears to me based on this feedback https://community-daily.mattermost.com/core/pl/7zgnwx1i1fy98f8rhbxun8syew that you must pass the user mention keys, so that the Markdown component within `Message` only highlight the mention keys of the user and not take into consideration mentions for other users in the post. See https://github.com/mattermost/mattermost-mobile/blob/gekidou/app/components/markdown/at_mention/index.tsx#L95 where mentionKeys is the prop being pass down

Reply:
mention keys gets passed down correctly

Reply:
where?

Reply:
in the Markdown component

Reply:
This issue was due to ldap groups, we were adding a mention key for each group's name, but group.name doesn't always exist.